qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDbConnector.groovy

```diff
@@ -334,9 +334,7 @@ class CustomerDbConnector implements CreateCustomerDataSource, UpdateCustomerDat
       try {
         category = CATEGORY_FACTORY.getForString(row.category as String)
       } catch (IllegalArgumentException ignored) {
-        //fixme this should not happen but there is an incomplete entry in the DB
         log.warn("Affiliation ${row.id} has category '${row.category}'. Could not match.")
-        category = AffiliationCategory.UNKNOWN
```

Comment:
I think we should not catch the exception, and leave it silent for the use case. The query or insert of an unknown affiliation category is a business rule violation.

I suggest to throw a DatabaseQueryException, and leave it to the use case how to deal with it.

Reply:
Good Point!